internal/controllers/provisioningrequest_setup.go

```diff
@@ -288,7 +289,7 @@ func (r *ProvisioningRequestReconciler) findManagedClusterForProvisioningRequest
 
 // findPoliciesForProvisioningRequests creates reconciliation requests for the ProvisioningRequests
 // whose associated ManagedClusters have matched policies Updated or Deleted.
-func findPoliciesForProvisioningRequests[T deleteOrUpdateEvent](
+func findPoliciesForProvisioningRequests[T deleteUpdateEvent](
```

Comment:
I wonder if we can use `EnqueueRequestsFromMapFunc` for policies as well since we only watch on delete and update events for policies.